upd/MaprDBUpdater.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"fmt"
+	"gopkg.in/urfave/cli.v2"
+	"log"
+	"os"
+	"sync"
+	"vmrenter/pkg/config"
+	"vmrenter/pkg/mapr"
+	"vmrenter/pkg/models"
+	"vmrenter/pkg/utils"
+)
+
+var nodesTable = "/user/mapr/nodes"
+
+func update(c *cli.Context) error {
+
+	if !c.IsSet("urldbconn") {
+		fmt.Println("Connection string isn't set, aborting")
+		return nil
+	}
+
+	if !c.IsSet("nodesfile"){
+		fmt.Println("File path string isn't set, aborting")
+		return nil
+	}
+	config.SetURLDBConn(c.String("urldbconn"))
+	dbConn := config.GetURLDBConn()
+	var csvFilePath = c.String("nodesfile")
+
+	fmt.Printf("**** Config **** %v\n", dbConn)
+
+	// Getting nodes id, ExpiresAt and ClusterID from /user/mapr/nodes table
+	fmt.Println("Starting getting nodes id, ExpiresAT, ClusterID...")
+	partialNodes := mapr.GetPartialReservationsForNodesUpdate()
+	err := mapr.Reset(nodesTable)
+	if err != nil {
+		fmt.Printf("Error occured while resetting /user/mapr/nodes table: %v", err)
+		return err
+	}
+	fmt.Println("Finished getting nodes id, ExpiresAT, ClusterID!")
+
+	// Getting nodes from csv file
+	fmt.Println("Starting getting nodes from the csv file...")
+	nodes := utils.GetNodesFromCSV(csvFilePath)
+	fmt.Println("Finished getting nodes from the csv file!")
+
+	listOfMaps := make([]map[string]interface{}, 0) // List of NodeDBJsons
+
+	// Creating NodeDBJsons from nodes
+	var wg sync.WaitGroup
+	nodeDbJsonQueue := make(chan map[string]interface{}, 1)
+	wg.Add(len(nodes))
+
+	fmt.Println("Starting creating NodeDBJsons from nodes...")
+	for _, node := range nodes {
+		go func(node models.Node) {
+			mapIntface := utils.GetNodeJsonDocMap(node)
+			mapIntface["_id"] = node.ID
+			nodeDbJsonQueue <- mapIntface
+		}(node)
+	}
+
+	go func() {
+		for n := range nodeDbJsonQueue {
+			listOfMaps = append(listOfMaps, n)
+			wg.Done()
+		}
+	}()
```

Comment:
If there is no plan on adding more than one thread (and I don't think it really should)
I think the code above is over complicated for what it does.
In the current state, there is no need for any thread or wait group.

Reply:
@sogwiz What do you think about that?

Reply:
@alaingilbert Fixed